controllers/infinispan_controller.go

```diff
@@ -301,21 +301,22 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 
 	// If an overlay configuration is specified wait for the related ConfigMap
 	overlayConfigMap := &corev1.ConfigMap{}
-	var overlayConfigMapKey string
+	var overlayConfigMapKey, overlayLog4jConfigMapKey string
 	if infinispan.Spec.ConfigMapName != "" {
 		if result, err := kube.LookupResource(infinispan.Spec.ConfigMapName, infinispan.Namespace, overlayConfigMap, infinispan, r.Client, reqLogger, r.eventRec, r.ctx); result != nil {
 			return *result, err
 		}
-		var foundKey bool
-		// Loop through the data looking for something like xml,json or yaml
+		// Loop through the data looking for something like xml, json or yaml
 		for overlayConfigMapKey = range overlayConfigMap.Data {
 			if overlayConfigMapKey == "infinispan-config.xml" || overlayConfigMapKey == "infinispan-config.json" || overlayConfigMapKey == "infinispan-config.yaml" {
-				foundKey = true
 				break
 			}
 		}
-		if !foundKey {
-			err := fmt.Errorf("infinispan-copnfig.[xml|yaml|json] configuration not found in ConfigMap: %s", overlayConfigMap.Name)
+		// Check if the user added a custom log4j.xml config
+		overlayLog4jConfigMapKey = overlayConfigMap.Data["log4j.xml"]
```

Comment:
No we're storing the contents of the log4j.xml file, so the var name `overlayLog4jConfigMapKey` doesn't make sense.

I would change the var to something like `var userLog4jConfig bool` , then you can just do: `_, userLog4jConfig = overlayConfigMap.Data["log4j.xml"]`.

The `buildStartupArgs` and other method signatures will then need to be updated to take a `bool` instead of a `string`.

Reply:
ack! Updated the code accordingly, thxs.